tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,19 +71,1106 @@ public function test_wp_enqueue_script() {
+		$this->assertSame( $expected_footer, $actual_footer, 'Scripts registered/enqueued using the older $in_footer parameter or the newer $args parameter should have the same outcome.' );
+	}
+
+	/**
+	 * Data provider for test_setting_in_footer_and_strategy.
+	 *
+	 * @return array[]
+	 */
+	public function get_data_for_test_setting_in_footer_and_strategy() {
+		return array(
+			// Passing in_footer and strategy via args array.
+			'async_footer_in_args_array'    => array(
+				'set_up'   => static function ( $handle ) {
+					$args = array(
+						'in_footer' => true,
+						'strategy'  => 'async',
+					);
+					wp_enqueue_script( $handle, '/footer-async.js', array(), null, $args );
+				},
+				'group'    => 1,
+				'strategy' => 'async',
+			),
+
+			// Passing in_footer=true but no strategy.
+			'blocking_footer_in_args_array' => array(
+				'set_up'   => static function ( $handle ) {
+					wp_register_script( $handle, '/defaults.js', array(), null, array( 'in_footer' => true ) );
+				},
+				'group'    => 1,
+				'strategy' => false,
+			),
+
+			// Passing async strategy in script args array.
+			'async_in_args_array'           => array(
+				'set_up'   => static function ( $handle ) {
+					wp_register_script( $handle, '/defaults.js', array(), null, array( 'strategy' => 'async' ) );
+				},
+				'group'    => false,
+				'strategy' => 'async',
+			),
+
+			// Passing empty array as 5th arg.
+			'empty_args_array'              => array(
+				'set_up'   => static function ( $handle ) {
+					wp_register_script( $handle, '/defaults.js', array(), null, array() );
+				},
+				'group'    => false,
+				'strategy' => false,
+			),
+
+			// Passing no value as 5th arg.
+			'undefined_args_param'          => array(
+				'set_up'   => static function ( $handle ) {
+					wp_register_script( $handle, '/defaults.js', array(), null );
+				},
+				'group'    => false,
+				'strategy' => false,
+			),
+
+			// Test backward compatibility, passing $in_footer=true as 5th arg.
+			'passing_bool_as_args_param'    => array(
+				'set_up'   => static function ( $handle ) {
+					wp_enqueue_script( $handle, '/footer-async.js', array(), null, true );
+				},
+				'group'    => 1,
+				'strategy' => false,
+			),
+
+			// Test backward compatibility, passing $in_footer=true as 5th arg and setting strategy via wp_script_add_data().
+			'bool_as_args_and_add_data'     => array(
+				'set_up'   => static function ( $handle ) {
+					wp_register_script( $handle, '/footer-async.js', array(), null, true );
+					wp_script_add_data( $handle, 'strategy', 'defer' );
+				},
+				'group'    => 1,
+				'strategy' => 'defer',
+			),
+		);
+	}
+
+	/**
+	 * Test setting in_footer and strategy.
+	 *
+	 * @dataProvider get_data_for_test_setting_in_footer_and_strategy
+	 * @ticket 12009
+	 * @covers ::wp_register_script
+	 * @covers ::wp_enqueue_script
+	 * @covers ::wp_script_add_data
+	 *
+	 * @param callable     $set_up            Set up.
```

Comment:
```suggestion
	 * @ticket 12009
	 *
	 * @covers ::wp_register_script
	 * @covers ::wp_enqueue_script
	 * @covers ::wp_script_add_data
	 *
	 * @dataProvider get_data_for_test_setting_in_footer_and_strategy
	 *
	 * @param callable     $set_up            Set up.
```

Apply same changes to other tests.

Reply:
Done in 83da501